server/mdm/maintainedapps/sync.go

```diff
@@ -91,6 +104,19 @@ func Refresh(ctx context.Context, ds fleet.Datastore, logger kitlog.Logger) erro
 		}
 	}
 
+	for _, a := range currentFMASet {
```

Comment:
Doing things this way is O(n^2), versus there where-not-in. I thought about handling things this way but the alternative seems cleaner since you can just provide the list of slugs to keep.